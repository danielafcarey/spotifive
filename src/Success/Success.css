.success {
  width: 45%;
  margin: 50px auto;
  text-align: center; }
  .success .artist-image {
    width: 300px;
    height: 300px;
    border-radius: 50%;
    object-fit: cover; }
  .success p {
    padding-top: 0; }
  .success ul {
    list-style: none;
    text-align: left; }
  .success .track {
    display: flex;
    flex-direction: row;
    align-items: center; }
    .success .track li {
      font-size: 16px;
      padding: 10px; }
