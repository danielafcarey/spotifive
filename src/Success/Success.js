import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

class Success extends Component {

  handleClick = (event) => {
    const { name } = event.target;
    this.props.changeRoute(name);
  }

  render() {
    const { name, image, topTracks, loggedIn } = this.props; 

    if (!loggedIn) {
      return <Redirect to='/' />;
    } 

    return (
      <div className='success' >
        <img 
          className='artist-image'
          src={ image } 
          alt={ name } />
        <h1>Success!</h1>
        <p>{ `${name} has been added to your Spotifive playlist` }</p>
        <button
          onClick={ this.handleClick } 
          name='search'>
          Search for another artist
        </button>
        <button
          onClick={ this.handleClick } 
          name='spotify'>
          Go to Spotify 
        </button>
      </div>  
    )

  }
}

export default Success;
