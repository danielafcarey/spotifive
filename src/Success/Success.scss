.success {
  width: 45%;
  margin: 50px auto;
  text-align: center;

  button {
    margin-top: 30px;
  }

  .artist-image {
    width: 300px;
    height: 300px;
    border-radius: 50%;
    object-fit: cover;
  }

  p {
    padding-top: 0;
  }

  ul {
    list-style: none;
    text-align: left;
  }

  .track {
    display: flex;
    flex-direction: row;
    align-items: center;

    li {
      font-size: 16px;
      padding: 10px;
    }
  }

}

