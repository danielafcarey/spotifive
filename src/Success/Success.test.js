import React from 'react';
import { shallow } from 'enzyme';
import Success from './Success';

describe('Success', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      name: 'The Rock',
      image: 'the rocks face',
      topTracks: ['i am rock', 'i love rock'],
      loggedIn: true
    }
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<Success { ...mockProps } />);

    expect(wrapper).toMatchSnapshot();
  })

})
