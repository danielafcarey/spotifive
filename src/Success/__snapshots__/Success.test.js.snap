// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Success
    image="the rocks face"
    loggedIn={true}
    name="The Rock"
    topTracks={
        Array [
            "i am rock",
            "i love rock",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="The Rock"
          className="artist-image"
          src="the rocks face"
/>,
        <h1>
          Success!
</h1>,
        <p>
          The Rock has been added to your Spotifive playlist
</p>,
        <button
          name="search"
          onClick={[Function]}
>
          Search for another artist
</button>,
        <button
          name="spotify"
          onClick={[Function]}
>
          Go to Spotify
</button>,
      ],
      "className": "success",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "The Rock",
          "className": "artist-image",
          "src": "the rocks face",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Success!",
        },
        "ref": null,
        "rendered": "Success!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Rock has been added to your Spotifive playlist",
        },
        "ref": null,
        "rendered": "The Rock has been added to your Spotifive playlist",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for another artist",
          "name": "search",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search for another artist",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go to Spotify",
          "name": "spotify",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Go to Spotify",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="The Rock"
            className="artist-image"
            src="the rocks face"
/>,
          <h1>
            Success!
</h1>,
          <p>
            The Rock has been added to your Spotifive playlist
</p>,
          <button
            name="search"
            onClick={[Function]}
>
            Search for another artist
</button>,
          <button
            name="spotify"
            onClick={[Function]}
>
            Go to Spotify
</button>,
        ],
        "className": "success",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "The Rock",
            "className": "artist-image",
            "src": "the rocks face",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Success!",
          },
          "ref": null,
          "rendered": "Success!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Rock has been added to your Spotifive playlist",
          },
          "ref": null,
          "rendered": "The Rock has been added to your Spotifive playlist",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for another artist",
            "name": "search",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search for another artist",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go to Spotify",
            "name": "spotify",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Go to Spotify",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
